Sdk improvements to support HFT service

njannink opened the pull request:
- Added options to configure additional services
- Added mechanism to load modules outside entry assembly
- Added option to register additional middleware
- Added Maintenancemode middleware
- Added default startup routine

njannink committed:
Sdk improvements for more complex services

- Upgraded Swagger to 2.4.0
- Added options to configure additional services
- Added mechanism to load modules outside entry assembly

njannink committed:
Rolled back to swagger v1.2

njannink committed:
Extend Lykke sdk

- Add option to register additional middleware
- Add Maintenancemode middleware
- Added default startup routine

KonstantinRyazantsev commented on src/Lykke.Sdk/LykkeServiceCollectionContainerBuilderExtensions.cs:
```diff
@@ -98,9 +98,14 @@ public static class LykkeServiceCollectionContainerBuilderExtensions
 
-            builder.RegisterInstance(configurationRoot).As<IConfigurationRoot>();            
+            builder.RegisterInstance(configurationRoot).As<IConfigurationRoot>();
+            builder.RegisterInstance(settings).As<IReloadingManager<TAppSettings>>();
```
Please, try to avoid settings rot registration in the container

KonstantinRyazantsev commented on src/Lykke.Sdk/LykkeStarter.cs:
```diff
@@ -0,0 +1,75 @@
+            where TStartup : class
+        {
+            Console.WriteLine($@"{componentName} version {
+                    Microsoft.Extensions.PlatformAbstractions.PlatformServices.Default.Application.ApplicationVersion
```
Lykke.Common.AppEnvironment could be used here

KonstantinRyazantsev commented on src/Lykke.Sdk/LykkeStarter.cs:
```diff
@@ -0,0 +1,75 @@
+                Console.WriteLine(@"Is RELEASE");
+            }
+
+            Console.WriteLine($@"ENV_INFO: {EnvInfo}");
```
Lykke.Common.AppEnvironment could be used here

KonstantinRyazantsev commented on src/Lykke.Sdk/Middleware/MaintenanceModeBuilder.cs:
```diff
@@ -0,0 +1,29 @@
+        /// <param name="builder">The builder.</param>
+        /// <param name="getMode">The maintenance mode getter.</param>
+        /// <returns>the builder</returns>
+        public static IApplicationBuilder UseMaintenanceMode<TAppSettings>(this IApplicationBuilder builder, Func<TAppSettings, MaintenanceMode> getMode)
```
Will be nice to add possibility to defer MaintenanceMode evaluation. This could be useful to switch the mode by API call without service restart.

njannink committed:
Maintenance mode support

- Processed review comments
- Maintenance mode checking is now on a timer

njannink committed:
Fix build and pack warnings

- Added xml comments
- Removed pdb from pack comment since its already packed by default
